### PR TITLE
Add ghc 9.0 to ci

### PR DESCRIPTION
**Pull request by fgaz**
Let's see if this works...

#7315

### CODE REVIEW

**Comment by fgaz** [comment]
I guess it has to be in master...

File: .github/workflows/linux.yml
Comment (phadej) [inline .github/workflows/linux.yml:23]: 9.0.1 job will start after 8.8.3 passes.

**Comment by phadej** [comment]
At least in `master`:

```
% make cabal-install-dev
% cabal build -w ghc-9.0.1 --dry all
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: Cabal-3.5.0.0 (user goal)
[__1] trying: base-4.15.0.0/installed-4.15.0.0 (dependency of Cabal)
[__2] trying: Cabal-described-3.5.0.0 (user goal)
[__3] next goal: rere (dependency of Cabal-described)
[__3] rejecting: rere-0.1 (conflict: base==4.15.0.0/installed-4.15.0.0, rere
=> base>=4.3.0.0 && <4.15)
[__3] fail (backjumping, conflict set: Cabal-described, base, rere)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, Cabal, Cabal-described, rere
Try running with --minimize-conflict-set to improve the error message.
```

So I doubt the job will succeed.

File: .github/workflows/linux.yml
Comment (fgaz) [reply .github/workflows/linux.yml:23]: Aha, thanks. I assumed it would show as "waiting for dependency" or something

Comment (phadej) [reply .github/workflows/linux.yml:23]: The *Waiting for status to be reported* jobs are there because they are `required` to pass before non-admins can merge. I.e. they come from repository settings, not in-repo CI config.

**Comment by fgaz** [comment]
it did fail, but even before than the solving phase: there's no 9.0 container

**Comment by phadej** [comment]
Hmm. I build the containers, but looks like I forgot to push them. They'll be up in an hour or so (many bytes to upload).

**Comment by fgaz** [comment]
Blocked on https://github.com/haskell-hvr/cryptohash-sha256/issues/12 (or I could leave there the allow-newer)

**Comment by phadej** [comment]
I pushed the containers for 9.0

**Comment by fgaz** [comment]
Blocked on https://hub.darcs.net/shelarcy/regex-compat-tdfa/issue/2 too

**Comment by phadej** [comment]
@fgaz, I'd drop `regex-compat-tdfa` from `cabal-testsuite` in favour of `regex-posix` or `regex-tdfa` (both of which work on windows nowadays).

**Comment by fgaz** [comment]
...I just copied the only function we use and fixed it.

@phadej what would you replace regexes with?

**Comment by phadej** [comment]
Good question. I'm not sure what `regex-compat-tdfa` does, looks like that not much, so I'd just use `regex-tdfa` https://hackage.haskell.org/package/regex-compat-tdfa-0.95.1.4/docs/src/Text-Regex.html

**Review by emilypi — APPROVED** [review]

**Pull request merged**
